tools/arxiv_downloader.py

```
@@ -0,0 +1,222 @@
+import re
+from tqdm import tqdm
+from database.DBEntity import AuthorMapping, AoP, PaperMapping
+from furnace.Author import Author
+from furnace.arxiv_paper import Arxiv_paper, get_arxiv_id_from_url
+from sqlalchemy import create_engine, Column, Integer, String
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+from furnace.google_scholar_paper import Google_paper
+from furnace.semantic_scholar_paper import S2paper
+from tools.PDF_generator import render_pdf
+from tools.gpt_util import *
```

Ruff (0.8.2)

`from tools.gpt_util import *` used; unable to detect undefined names

(F403)